src/unit_tests/os_auth/test_auth_parse.c

```diff
@@ -81,7 +81,7 @@ parse_evaluator parse_values_default_cfg []={
     { "OSSEC PASS: pass124 OSSEC A:'agent0'", "192.0.0.1", NULL,        {NULL, NULL, NULL}, {OS_INVALID,"ERROR: Invalid request for new agent"}, {"Invalid request for new agent from: 192.0.0.1", NULL, NULL, NULL} },
     { "OSSEC A:''", "192.0.0.1", NULL,                                  {NULL, NULL, NULL}, {OS_INVALID,"ERROR: Invalid agent name: "},          {"Invalid agent name:  from 192.0.0.1", NULL, "Received request for a new agent () from: 192.0.0.1", NULL} },
     { "OSSEC A:'inv;agent'", "192.0.0.1", NULL,                         {NULL, NULL, NULL}, {OS_INVALID,"ERROR: Invalid agent name: inv;agent"}, {"Invalid agent name: inv;agent from 192.0.0.1", NULL, "Received request for a new agent (inv;agent) from: 192.0.0.1", NULL} },
-    
+
```

Comment:
To test the correct parsing of the key hash we should add key_hash to the **_enrollment_param** structure.

Reply:
To test the new correct parsing of the new key_hash we should add here a new item in parse_values_default_cfg like:
`{ "OSSEC A:'agent1' K:'AaAa", "192.0.0.1", NULL,                            {"192.0.0.1", "agent1", NULL, "AaAa"},              {OS_SUCCESS,""}, {NULL, NULL, "Received request for a new agent (agent1) from: 192.0.0.1", NULL} },`

Reply:
Now the tests covers more cases